src/cargo/ops/cargo_compile.rs

```diff
@@ -98,6 +99,48 @@ pub fn compile(manifest_path: &Path,
     compile_pkg(&package, options)
 }
 
+pub fn resolve_dependencies<'a, 'b>(package: &Package, config: &'a Config<'b>,
```

Comment:
Is this `'b` necessary? It looks like it's only used in `Config` so you may be able to remove it.